.github/workflows/pull_request.yml

```diff
@@ -58,7 +58,7 @@ jobs:
       - name: Setup Golang
         uses: actions/setup-go@v2
         with:
-          go-version: '1.19.1'
+          go-version: '1.18.7'
```

Comment:
@liranr23 @bennyz @ahadas 
This raises the question of which go we would use/test.
The `go.mod` has defined the 1.14 but the CI was running on 1.19.1 on which the migration failed with:
```
go vet ./pkg/... ./cmd/...
# github.com/konveyor/forklift-controller/pkg/lib/inventory/model
Error: pkg/lib/inventory/model/client.go:557:6: second argument to errors.As should not be *error
Error: pkg/lib/inventory/model/client.go:650:6: second argument to errors.As should not be *error
make: *** [Makefile:55: vet] Error 2
Error: Process completed with exit code 2.
```
I was running the migration on my env with 1.18.7 and the ci passed so that's why I changed it. It is possible that I might just need to bump my version of go to 1.19.1 but we should decide which version to use.

Reply:
IMO we do need to *try* and have the latest go version. however, if it breaks something it's ok to have it with older version and have a task to upgrade golang version? @ahadas

Reply:
I agree with Liran. It's fine to merge this with older version but we should at least have a task to try to fix the problem and upgrade to latest Go.

Also, note that Bazel `WORKFLOW` has: `go_register_toolchains(version = "1.16.5")`
This is the version that we're actually using. I believe that this should be in sync with what you have for CI.

I quickly tried to build with bazel for both 1.18.7 and 1.19.1 the first builds fine but for the later it fails with:

```
2022/11/08 10:19:14 link: unlinkable object (from package github.com/konveyor/forklift-controller/vendor/github.com/cespare/xxhash/v2) - compiler requires -p flag
```

So we have another problem to solve. As for the problem with `errors.As` from a quick look we're using it wrong in the code. It should either be used in the form:
```
var notFound *NotFound;
if errors.As(err, notFound) {
```

or, which is probably preferable in our case since we don't need the value, use `errors.Is` instead.

Reply:
Btw 1.19.1 is not the latest anymore. There are already 1.19.2 (released in October) and 1.19.3 (released in November).

Reply:
The suggestion to split it to two lines did not help.
So I updated it according to the docs as in [1]
```
if err == NotFound{
```
[1] http://go-database-sql.org/errors.html